public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx

```diff
@@ -66,6 +100,16 @@ export const DBClusterAdvancedOptions: FC<FormRenderProps> = ({ values, form })
     setPrevResources(resources);
   }, [resources]);
 
+  useEffect(() => {
+    if (kubernetesCluster) {
+      getResources();
+
+      timer = setInterval(() => getResources(false), RECHECK_INTERVAL);
```

Comment:
I know the interval is quite large for this to happen, but this is calling setInterval every `RECHECK_INTERVAL
` ms even if the previous API call is pending. Would it be worth to instead call a function that waits for `getResources`
 and then calls `setTimeout` to recursively call itself?

Reply:
Hmm I don't know, because even if the call is pending I want to schedule another one after 10s. If one call takes more time most likely the others would do too. This would only be a problem if a new call completes first than an old one, but I don't see that happening honestly. That solution has a drawback too, if one call won't complete we have to wait until it timesout to trigger a new one. Anyway I think your solution is still more reliable, but I will do some tests and implement it in another PR.

Reply:
Sounds like a plan 🚀